File: AdventOfCode2023/day1-1.go
package main

import (
	"bufio"
	"os"
	"strconv"
	"unicode"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	vals := make([]int, 0)
	for {
		line, err := reader.ReadString('\n')
		if err != nil {
			break
		}

		first := ""
		last := ""
		for _, ch := range line {
			if unicode.IsDigit(rune(ch)) {
				if first == "" {
					first = string(ch)
				}
				last = string(ch)
			}
		}
		first += last
		x, _ := strconv.Atoi(first)
		vals = append(vals, x)
	}
	sum := 0
	for _, v := range vals {
		sum += v
	}
	println("Ans:", sum)
}

File: AdventOfCode2023/day1-1.rs
use std::io;

fn main() {
    let mut ans = 0;
    for line in io::stdin().lines() {
        let mut first: i32 = -1;
        let mut last: i32 = -1;
        for ch in line.unwrap().chars() {
            if ch.is_digit(10) {
                if first == -1 {
                    first = ch.to_digit(10).unwrap() as i32;
                }
                last = ch.to_digit(10).unwrap() as i32;
            }
        }
        ans += (first*10)+last;
    }
    println!("Ans: {ans}")
}
